docs/guide/plugins.md

```diff
@@ -0,0 +1,202 @@
+
+![cli plugin dev mode](./../assets/cli-plugin-dev-mode.png)
+
+It shows then a button in the footer, which allows us to turn on Dev mode, and see all plugin slots in theme.
+
+![turn-on-dev-mode](./../assets/turn-on-dev-mode.png)
+
+it will show us available slots on the theme, and you can click on any of them to see its name in the console
+
+![theme-slots](./../assets/theme-slots.png)
+
+### Local plugin
+
+Let's create `cool-promotion-plugin`, which can display our special product under the top header.
+We're creating following files inside your project.
+
+![local-plugin-structure](./../assets/local-plugin-structure.png)
+
+```
+.
+├── sw-plugins                  <- Local folder for developing and overriding plugins
+│   ├── local-plugins.json      <- Config file to manage the order of plugins
+│   ├── cool-promotion-plugin   <- your local plugin directory, maybe more of them
+│   │   ├── config.json         <- plugin configuration
+│   │   └── *.vue               <- vue files for your plugin
+│   ├── another-local-plugin
+│   │   ├── config.json
+│   │   └── *.vue
+```
+
+so in `local-plugins.json` we're activating our new plugin
+
+```json
+{
+  "cool-promotion-plugin": true
+}
+```
+
+inside `cool-promotion-plugin/config.json` we need to put
+
+- slot, in which we want to inject
+- vue file, which should be displayed there
+
+In the single plugin, we may inject in multiple plugin slots. Let's inject our `main.vue` file into `top-header-after` slot.
+
+```json{4,5}
+{
+  "slots": [
+    {
+      "name": "top-header-after",
+      "file": "main.vue"
+    }
+  ],
+  "settings": {}
+}
+```
+
+so our `main.vue` file can be as simple as
+
+```vue
+<template>
+  <div class="cool-promotion-plugin">
+    <h3>Hello from plugin!</h3>
+  </div>
+</template>
+<script>
+export default {
+  name: "CoolPromotionPlugin",
+  data() {
+    return {};
+  },
+};
+</script>
+<style lang="scss" scoped>
+.cool-promotion-plugin {
+  display: flex;
+  justify-content: center;
+  border: 1px solid green;
+}
+</style>
+```
+
+we run `shopware-pwa plugins`, and we see the effect
+
+![local-plugin-1](./../assets/local-plugin-1.png)
+
+Let's put some logic there, and we're getting the cheapest product and displaying it with link.
+
+```vue
+<template>
+  <div class="cool-promotion-plugin">
+    <!-- we can link to our product product -->
+    <nuxt-link
+      :to="getProductLink"
+      v-if="product"
+      class="cool-promotion-plugin__link"
+    >
+      <h3>{{ getName }} in special price! {{ getPrice | price }}</h3>
+    </nuxt-link>
+  </div>
+</template>
+<script>
+import { getProducts } from "@shopware-pwa/shopware-6-client";
+import {
+  getProductName,
+  getProductRegularPrice,
+  getProductUrl,
+} from "@shopware-pwa/helpers";
+
+export default {
+  name: "CoolPromotionPlugin",
+  data() {
+    return {
+      product: null,
+    };
+  },
+  async mounted() {
+    // let's get the cheapest product
+    const productsResult = await getProducts({
```

Comment:
Maybe `{ productResult } = await...`?

Reply:
we cannot do that, getProducts returns `SearchResult<Product[]>` so 
```
{
 total: number;
 data: Product[]
}
```
we could do
```
const {data} = await ...
```
but I'm not sure we should do this in example :)

Reply:
That's what I meant, my mistake! :) I thought it could be more concise, but I'll leave that up to you.

Reply:
I know it's only an example piece of code but we should always remember to catch the exceptions that may be thrown by the client :smile: otherwise bad things happen in the runtime :) It's related to upcoming notification system (not ready in UI) for asynchronous requests... just saying here, but maybe it's not the best place to talk about.